app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt

```diff
@@ -45,34 +57,69 @@ class Settings : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_settings)
 
+        instantiateViewModel()
+
         supportActionBar?.hide()
 
         val emailAddressText = findViewById<TextView>(R.id.email_address_text)
-        emailAddressText.text = user.getEmail() ?: getString(R.string.not_logged_in)
+        emailAddressText.text = userHelper.getEmail() ?: getString(R.string.not_logged_in)
+
+        val location = findViewById<TextView>(R.id.current_location_text)
+
+        val radius = findViewById<TextView>(R.id.radius_text)
+        radiusSlider = findViewById(R.id.location_slider)
 
-        val locationText = findViewById<TextView>(R.id.current_location_text)
-        locationText.text = getString(R.string.lausanne_switzerland)
+        val ageRange = findViewById<TextView>(R.id.selected_age_range_text)
+        ageRangeSlider = findViewById(R.id.age_range_slider)
 
-        val radiusText = findViewById<TextView>(R.id.radius_text)
-        val radiusSlider = findViewById<Slider>(R.id.location_slider)
+        val showMe = findViewById<TextView>(R.id.show_me_text)
+
+        //Retrieve settings from database
+        viewModel.user.observe(this) {
+            location.text = AndroidLocationService.getCurrentLocation(this, it)
+            currentRadius = it.radius!!
```

Comment:
I think that we should either have radus in User non-nullable, or use the Elvis operator the use a default like `it.radius ?: DEFAULT_RADIUS`. Having null values crash the app always seems dangerous to me, especially since the problem we had yesterday.